pxr/usdImaging/usdImaging/lightAdapter.cpp

```diff
@@ -71,15 +71,50 @@ UsdImagingLightAdapter::Populate(UsdPrim const& prim,
 {
     index->InsertSprim(HdPrimTypeTokens->light, prim.GetPath(), prim);
     HD_PERF_COUNTER_INCR(UsdImagingTokens->usdPopulatedPrimCount);
+    _RegisterLightCollections(prim);
 
     return prim.GetPath();
 }
 
 void
-UsdImagingLightAdapter::_RemovePrim(SdfPath const& cachePath,
-                                    UsdImagingIndexProxy* index)
+UsdImagingLightAdapter::MarkCollectionsDirty(UsdPrim const& prim,
+                                             SdfPath const& cachePath,
+                                             UsdImagingIndexProxy* index)
+{
+    index->MarkSprimDirty(cachePath, HdLight::DirtyCollection);
+}
+
+bool
+UsdImagingLightAdapter::_UpdateCollectionsChanged(UsdPrim const& prim) const {
+    UsdImaging_CollectionCache &collectionCache = _GetCollectionCache();
+    UsdLuxLightAPI light(prim);
+    bool lightColChanged = collectionCache.UpdateCollection(light.GetLightLinkCollectionAPI());
+    bool shadowColChanged = collectionCache.UpdateCollection(light.GetShadowLinkCollectionAPI());
+    return lightColChanged || shadowColChanged;
+}
+
+void
+UsdImagingLightAdapter::_UnregisterLightCollections(SdfPath const& cachePath) {
+    UsdImaging_CollectionCache &collectionCache = _GetCollectionCache();
+    SdfPath lightLinkPath = cachePath.AppendProperty(UsdImagingTokens->collectionLightLink);
+    collectionCache.RemoveCollection(_GetStage(), lightLinkPath);
+    SdfPath shadowLinkPath = cachePath.AppendProperty(UsdImagingTokens->collectionShadowLink);
+    collectionCache.RemoveCollection(_GetStage(), shadowLinkPath);
+}
+
+void
+UsdImagingLightAdapter::_RegisterLightCollections(UsdPrim const& prim) {
+    _UpdateCollectionsChanged(prim);
+}
+
+bool
+UsdImagingLightAdapter::_UpdateCollectionsChanged(UsdPrim const& prim) const {
```

Comment:
Probably just a merge issue, but _UpdateCollectionsChanged shows up twice in this file (seemingly with the same implementation).